x-pack/plugin/logsdb/src/main/java/org/elasticsearch/xpack/logsdb/LogsdbIndexModeSettingsProvider.java

```diff
@@ -196,25 +208,36 @@ boolean newIndexHasSyntheticSourceUsage(
                 // to disabled, because that isn't allowed with logsdb/tsdb. In other words setting index.mapping.source.mode setting to
                 // stored when _source.mode mapping attribute is stored is fine as it has no effect, but avoids creating MapperService.
                 var sourceMode = SourceFieldMapper.INDEX_MAPPER_SOURCE_MODE_SETTING.get(tmpIndexMetadata.getSettings());
-                return sourceMode == SourceFieldMapper.Mode.SYNTHETIC;
+                hasSyntheticSourceUsage = sourceMode == SourceFieldMapper.Mode.SYNTHETIC;
+                if (IndexSortConfig.INDEX_SORT_FIELD_SETTING.get(indexTemplateAndCreateRequestSettings).isEmpty() == false) {
+                    // Custom sort config, no point for further checks on [host.name] field.
+                    return new MappingData(hasSyntheticSourceUsage, false);
+                }
+                if (IndexSettings.LOGSDB_SORT_ON_HOST_NAME.get(indexTemplateAndCreateRequestSettings)) {
+                    // Setting for sorting on [host.name] is already injected, propagate it.
+                    return new MappingData(hasSyntheticSourceUsage, true);
+                }
             }
 
-            // TODO: remove this when _source.mode attribute has been removed:
             try (var mapperService = mapperServiceFactory.get().apply(tmpIndexMetadata)) {
                 // combinedTemplateMappings can be null when creating system indices
                 // combinedTemplateMappings can be empty when creating a normal index that doesn't match any template and without mapping.
                 if (combinedTemplateMappings == null || combinedTemplateMappings.isEmpty()) {
                     combinedTemplateMappings = List.of(new CompressedXContent("{}"));
                 }
                 mapperService.merge(MapperService.SINGLE_MAPPING_NAME, combinedTemplateMappings, MapperService.MergeReason.INDEX_TEMPLATE);
-                return mapperService.documentMapper().sourceMapper().isSynthetic();
+                return new MappingData(
+                    hasSyntheticSourceUsage || mapperService.documentMapper().sourceMapper().isSynthetic(),
+                    IndexSettings.LOGSDB_SORT_ON_HOST_NAME.get(indexTemplateAndCreateRequestSettings)
+                        || mapperService.mappingLookup().getMapper("host.name") instanceof FieldMapper
```

Comment:
I think an important consequence of this change is that sorting on `host.name` does not work in case the field is mapped dynamically. As we've switched to dynamic mappings on various integrations, we'll need to evaluate the impact of that.

Reply:
Also, according to the [docs](https://www.elastic.co/guide/en/elasticsearch/reference/current/index-modules-index-sorting.html)
> Only boolean, numeric, date and keyword fields with doc_values are allowed here.

Do we need to validate that the mapping for `host.name` conforms to  these constraints? A test case for this would be helpful, too.

Reply:
Maybe, instead of never injecting `host.name`, we could inject it in case it wouldn't cause conflicts and if the field doesn't exist already.